README.md

```diff
@@ -154,9 +140,55 @@ lspconfig.util.default_config = vim.tbl_extend(
 )
 ```
 
+## Individual server settings and initialization options
+
+See [CONFIG.md](CONFIG.md) for documentation and configuration of individual language servers.
+This document contains installation instructions for each language server, and is 
+auto-generated from the documentation in the lua source. Do not submit PRs 
+modifying CONFIG.md directly; CONFIG.md will be overwrriten by docgen
+
+**You do not need to copy settings or init_options from this configuration into your config**
+
+## Keybindings and completion
+
+The following maps most of the standard functions to keybindings, and maps omnicomplete to use
+the lsp.omnifunc. See `:help lsp for more details`
+```vim
+lua << EOF
+local nvim_lsp = require('lspconfig')
+local on_attach = function(_, bufnr)
+  local function buf_set_keymap(...) vim.api.nvim_buf_set_keymap(bufnr, ...) end
+  local function buf_set_option(...) vim.api.nvim_buf_set_option(bufnr, ...) end
+
+  buf_set_option('omnifunc', 'v:lua.vim.lsp.omnifunc')
+
+  -- Mappings.
+  local opts = { noremap=true, silent=true }
+  buf_set_keymap('n', 'gD', '<Cmd>lua vim.lsp.buf.declaration()<CR>', opts)
+  buf_set_keymap('n', 'gd', '<Cmd>lua vim.lsp.buf.definition()<CR>', opts)
+  buf_set_keymap('n', 'K', '<Cmd>lua vim.lsp.buf.hover()<CR>', opts)
+  buf_set_keymap('n', 'gi', '<cmd>lua vim.lsp.buf.implementation()<CR>', opts)
+  buf_set_keymap('n', '<C-k>', '<cmd>lua vim.lsp.buf.signature_help()<CR>', opts)
+  buf_set_keymap('n', '<space>wa', '<cmd>lua vim.lsp.buf.add_workspace_folder()<CR>', opts)
+  buf_set_keymap('n', '<space>wr', '<cmd>lua vim.lsp.buf.remove_workspace_folder()<CR>', opts)
+  buf_set_keymap('n', '<space>wl', '<cmd>lua print(vim.inspect(vim.lsp.buf.list_workspace_folders()))<CR>', opts)
+  buf_set_keymap('n', '<space>D', '<cmd>lua vim.lsp.buf.type_definition()<CR>', opts)
+  buf_set_keymap('n', '<space>rn', '<cmd>lua vim.lsp.buf.rename()<CR>', opts)
+  buf_set_keymap('n', 'gr', '<cmd>lua vim.lsp.buf.references()<CR>', opts)
+  buf_set_keymap('n', '<space>e', '<cmd>lua vim.lsp.diagnostic.show_line_diagnostics()<CR>', opts)
+  buf_set_keymap('n', '[d', '<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>', opts)
+  buf_set_keymap('n', ']d', '<cmd>lua vim.lsp.diagnostic.goto_next()<CR>', opts)
+  buf_set_keymap('n', '<space>q', '<cmd>lua vim.lsp.diagnostic.set_loclist()<CR>', opts)
```

Comment:
Maybe it would be a good idea to add an example of how to make a conditional keymapping depending on the resolved capabilities (ref GH-493)?

```lua
if client.resolved_capabilities.document_formatting then
  buf_set_keymap("n", "<Leader>f", "<cmd>lua vim.lsp.buf.formatting()<CR>", opts)
end
```

And document highlighting:

```lua
if client.resolved_capabilities.document_highlight then
  util.nvim_multiline_command [[
    augroup lsp_document_highlight
      autocmd!
      autocmd CursorHold <buffer> lua vim.lsp.buf.document_highlight()
      autocmd CursorMoved <buffer> lua vim.lsp.buf.clear_references()
    augroup END
  ]]
end
```

Reply:
omg why is nvim_multiline_command not in core.

Reply:
what? I didn't know we have `nvim_multiline_command`, that would be so useful 😆

Reply:
It's only in `lspconfig`, yet another reason to prefer `lspconfig` over CoC (just kidding)

Reply:
Also this reminds me to port the function signature of this version of on_attach to core.

Reply:
> omg why is nvim_multiline_command not in core.

Good question. Writing an `augroup` using `vim.cmd` isn't super pretty.

Reply:
What servers actually support document highlight? I can't test it.

Reply:
> What servers actually support document highlight? I can't test it.

Pyright and bash-language-server at least. Note that `CursorHold` can be pretty slow depending on your configuration since the default for `'updatetime'` is `4000`.

Reply:
Ah, then we may want to add notes about these (mine doesn't seem to, onedark)

Reply:
https://github.com/neovim/neovim/pull/13614

Reply:
Thanks! When merged I can add a reference